src/test/java/igym/services/WorkoutServiceTest.java

```diff
@@ -246,4 +255,64 @@ void testDeleteWorkoutWithNoExercises() {
         verify(workoutRepository).save(workout);
     }
 
+    @Test
+    @DisplayName("Test soft delete of existing exercise in workout")
+    void testDeleteExerciseByIdSuccess() {
+        UUID exerciseId = UUID.randomUUID();
+
+        Exercise ex1 = new Exercise();
+        ex1.setName("Squat");
+        ex1.setWeight(60);
+        ex1.setNumReps(10);
+        ex1.setNumSets(4);
+        ex1.setStatus(Status.active);
+        ReflectionTestUtils.setField(ex1, "id", exerciseId);
+
+        when(exerciseRepository.findByIdAndStatus(exerciseId, Status.active)).thenReturn(java.util.Optional.of(ex1));
+
+        workoutService.deleteExercise(exerciseId);
+
+        assertEquals(Status.inactive, ex1.getStatus());
+    }
+
+    @Test
+    @DisplayName("Test delete exercise when exercise not found in workout throws exception")
+    void testDeleteExerciseNotFoundInWorkout() {
+        UUID exerciseId = UUID.randomUUID();
+        UUID nonExistentExerciseId = UUID.randomUUID();
+
+        Exercise ex1 = new Exercise();
+        ex1.setName("Squat");
+        ex1.setWeight(60);
+        ex1.setNumReps(10);
+        ex1.setNumSets(4);
+        ex1.setStatus(Status.active);
+        ReflectionTestUtils.setField(ex1, "id", exerciseId);
+
```

Comment:
You forgot to mock the return. The test is passing because of an false positive scenario.

```suggestion
     when(exerciseRepository.findByIdAndStatus(nonExistentExerciseId, Status.active)).thenReturn(Optional.empty());
```